File: AD-HOC/1366/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n;
        while ((n = scanner.nextInt()) != 0) {
            int pairs = 0;
            for (int i = 0; i < n; i++) {
                scanner.nextInt();
                pairs += scanner.nextInt() / 2;
            }
            int rectangles = pairs / 2;
            System.out.println(rectangles);
        }
        scanner.close();
    }
}

File: AD-HOC/1366/main.c
#include <stdio.h>

int main() {
    int n, c, v;
    scanf("%d", &n);
    while (n != 0) {
        int pairs = 0;
        for (int i = 0; i < n; i++) {
            scanf("%d %d", &c, &v);
            pairs += v / 2;
        }
        int rectangles = pairs / 2;
        printf("%d\n", rectangles);
        scanf("%d", &n);
    }
    return 0;
}
